fix(mobile): remove log message counter

rovo89 opened the pull request:
Previously, the items in the log page were numbered starting with `#0` and increasing from top (newest) to bottom (oldest). Being new to the app, this confused me because I would have expected that newer messages have a higher number than older messages.

Initially, I had planned to just start counting from the bottom (i.e. the oldest message would be `#0`), but I guess the auto-increment ID of the message would be even better as it isn't reused after clearing the logs. That is, if a number is needed at all...

Since the numbers are no longer related to the total log count, I removed the same from the title. Instead, I added the message ID to the details page title.

shenlong-tanwen commented:
I'm okay with changing the order of the logs, but using the autoIncrement ID from Isar results in the following scenario where if you open the log page immediately, it will display a very large ID until you reopen the log page and then it would be fine.

<img width="344" alt="image" src="https://github.com/immich-app/immich/assets/139912620/d728b22d-568d-44ec-ac52-10e23f6d3990">

Can you check if you can reproduce this from your end as well?

shenlong-tanwen commented:
> If you have any preference, please let me know, otherwise I'd give option 4 a shot.

I don't have an actual preference but won't 3 be actually easier to implement? We already have the total number of logs and the current index, and so, it is just a matter of taking a difference between them and displaying it instead of directly displaying the index?

rovo89 commented:
Removing the index completely would be even easier as it's just about deleting some (parts of) lines. 😉

shenlong-tanwen commented:
@alextran1502 / @fyfrey Thoughts on how we can handle the log message numbering in the log page?

fyfrey commented:
We might just not show any number at all. Logs have a timestamp and their message+data. That should be enough.

rovo89 committed:
fix(mobile): remove log message counter

Previously, the items in the log page were numbered starting with `#0`
and increasing from top to bottom. Being new to the app, this confused
me because I would have expected that newer messages have a higher
number than older messages.

Removing the counter completely because it doesn't add any value -
log messages are identified by their timestamp, text and other details.

shenlong-tanwen commented:
@rovo89 We can remove the numbering completely. Can you update your PR with the required change?

rovo89 commented:
Sure, I have already done that locally. Will try tonight if moving the timestamp to the front looks better, then I'll push it.

rovo89 committed:
Switch timestamp and context in log overview

rovo89 commented:
I've change the subtitle to "at <timestamp> in <context>" to make the timestamp at little bit more prominent (and because it has a pretty much stable width across messages). I have some further ideas, but I think I'll save them for another PR.

shenlong-tanwen reviewed (approved):
LGTM!

shenlong-tanwen commented:
@rovo89 Can you re-base your changes over the latest main branch?

rovo89 commented:
Sure, will do that later today.

rovo89 commented:
Looks like @jrasm91 already rebased, thanks! I'm a bit confused though. Aren't you guys using squash merges anyway, so why rebase when there's no conflict?

jrasm91 commented:
> Looks like @jrasm91 already rebased, thanks! I'm a bit confused though. Aren't you guys using squash merges anyway, so why rebase when there's no conflict?

In this case there are github action workflow changes on main that were missing in the branch. Required actions weren't passing because they didn't exist. Rebase was to make sure the required actions actually passed with your code changes.
